RecoLocalCalo/HGCalRecAlgos/plugins/alpaka/HGCalRecHitCalibrationESProducer.cc

```diff
@@ -68,6 +68,18 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
         std::copy(begin, end, &column_SoA[offset]);
       }
 
+      template <typename T, typename P>
+      static void fill_SoA_eigen_row(P& soa, const std::vector<std::vector<T>>& values, const size_t row) {
+        if (row >= values.size())
+          throw edm::Exception(edm::errors::LogicError, "HGCalCalibrationESProducer")
+              << " Tried to copy row " << row << " to SoA, but only have " << values.size() << " values in JSON!";
+        if (!values.empty() && int(values[0].size()) != soa.size())
+          throw edm::Exception(edm::errors::LogicError, "HGCalCalibrationESProducer")
+              << " Expected " << soa.size() << " elements in Eigen vector, but got " << values[0].size() << "!";
```

Comment:
Shouldn't this be
```suggestion
        if (!values.empty() && int(values[row].size()) != soa.size())
          throw edm::Exception(edm::errors::LogicError, "HGCalCalibrationESProducer")
              << " Expected " << soa.size() << " elements in Eigen vector, but got " << values[row].size() << "!";
```
?

Or all the elements of the class `T` are expected to have the same size by construction?